src/clients/cassandra/CassandraConnector.js

```diff
@@ -1,17 +1,75 @@
 'use strict';
 
 const Promise = require('promise');
+const cassandra = require('cassandra-driver');
+const cassandraTableStorageManager = require('../storage/CassandraTableStorageManager');
+
+/** Default Cassandra Client Options
+ * https://github.com/datastax/nodejs-driver/blob/master/lib/client-options.js
+ * 
+ * Cassandra Client Configurable Options
+ * https://docs.datastax.com/en/developer/nodejs-driver/3.2/api/type.ClientOptions/
+ * 
+ * Since datastax node driver does not have much detail on pooling, the java driver doc will be used:
+ * http://docs.datastax.com/en/developer/java-driver/3.3/manual/pooling/
+ * 
+ * Connection pools have a variable size, which gets adjusted automatically depending on the current load. 
+ * There will always be at least the core number of connections.
+ * 
+ * Connection pool picks the connection with the minimum number of in-flight requests.
+ * 
+ * Cassandra version 2.1 or greater can send up to 32768 requests (stream ids) per connection.
+ * 
+ * CORE_CONNECTIONS_PER_HOST: to tune this option see java-driver 3.3 manual: 'Tuning protocol v3 for very high throughputs'
+ * 
+ * TODO: consider sslOptions and authProvider
+ */
+const distance = cassandra.types.distance;
+const CORE_CONNECTIONS_PER_HOST = 1;
+const options = {
+  contactPoints: [process.env.CASSANDRA_CONTACT_POINTS],
+  keyspace: process.env.CASSANDRA_KEYSPACE,
+  pooling: {
+    coreConnectionsPerHost: {
+      [distance.local]: CORE_CONNECTIONS_PER_HOST,
+      [distance.remote]: CORE_CONNECTIONS_PER_HOST
+    } 
+  }
+};
+
+/** Code should share the same Client instance across the application i.e.
```

Comment:
Comment is implied by the fact that you're instantiating this as a const and wrapping it with a custom client interface. Would thus avoid the comment as comments, like code,
 are a maintenance/comprehension cost (feel free to put the details in the commit message or pull request message though so that folks can refer to it in the future if more details are needed). You can use `git rebase -i master` to amend commit messages that are in the past.